### PR TITLE
Doc improvements

### PR DESCRIPTION
**Pull request by raarts**
As a beginner I spent many hours and asked questions on why the first example of react-navigation didn't work. This doc change fixes that.

**Test plan (required)**

This is a doc change

**Code formatting**

### CODE REVIEW

**Comment by matthamil** [comment]
Thank you for your contribution! I see documentation updates as a high priority currently with the number of open issues that are due to inadequate docs. I will review these documentation changes and provide any feedback to you within the next day or two. Feel free to tag me if I do not get back with you in time.

File: docs/guides/Guide-Intro.md [hunk 1]
```diff
@@ -51,10 +51,11 @@ class HomeScreen extends React.Component {
   }
 }
 
-const SimpleApp = StackNavigator({
+export default SimpleApp = StackNavigator({
   Home: { screen: HomeScreen },
 });
 
+// skip this line if using Create React Native App
```
Comment (matthamil) [inline docs/guides/Guide-Intro.md]: I think it might be best to have two "Hello World" examples for the absolute beginner. There was a [recent issue](https://github.com/react-community/react-navigation/issues/1945) where someone could not get the Hello World example running by copying and pasting this example into their text editor. I am in favor of having two code snippets: one for the `react-native` CLI and another tailored for `create-react-native-app` with a brief explanation _why_ the two snippets are different. A lot of beginners jump straight into using libraries like this one, and we want to minimize the amount of potential confusion that they might have.

File: docs/guides/Contributors.md [hunk 2]
```diff
@@ -22,6 +22,7 @@ npm install
 ### Run the example app
 
 ```
+npm install
```
Comment (gabrielbull) [inline docs/guides/Contributors.md]: Why did you add a npm install here?

**Comment by matthamil** [comment]
Have you made any progress on the requested changes @raarts ? Someone else can jump in on this PR if you do not have time to continue on it.

File: docs/guides/Guide-Intro.md (hunk 1 above)
Comment (jamesg1) [inline docs/guides/Guide-Intro.md]: As someone new to React Native that would remove the initial pain points of getting up and running. Thanks @matthamil

**Comment by raarts** [comment]
I am sorry, I really would like to extend the docs (I have written documentation on other products), but I feel I don't have enough understanding myself. I am still struggling with questions like: "why are navigators needed, why can't I just link screens together with some goto of sorts. I understand screens need a stack to go back, but stacks are really simple, so why make complete components for this. I understand there are general concepts like the screen with a header with an arrow pointing back, but why not provide a component to show that header as a separate component, instead of complicating things with attributes on screens. Also what makes this particular implementation better than others." 

This is not a rant by the way, even though it may sound like one. I'm trying to convey the things I'm confused with as a beginner. Thanks for listening.

**Comment by spencercarli** [comment]
Hi! This was completed in #2491
